src/Transports/MassTransit.ActiveMqTransport/Topology/Configuration/Specifications/ActiveMqBindConsumeTopologySpecification.cs

```diff
@@ -0,0 +1,82 @@
+namespace MassTransit.ActiveMqTransport.Topology.Specifications
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Builders;
+    using Configurators;
+    using Entities;
+    using GreenPipes;
+
+
+    /// <summary>
+    /// Used to by a Consumer virtual destination to the receive endpoint, via an additional message consumer
+    /// </summary>
+    public class ActiveMqBindConsumeTopologySpecification :
+        TopicBindingConfigurator,
+        IActiveMqBindingConsumeTopologySpecification
+    {
+        //readonly string _consumerName;
+
+        public ActiveMqBindConsumeTopologySpecification(string topicName, bool durable = true, bool autoDelete = false)
+            : base(topicName, durable, autoDelete)
+        {
+            //_consumerName = consumerName;
+        }
+
+        public IEnumerable<ValidationResult> Validate()
+        {
+            return Enumerable.Empty<ValidationResult>();
+        }
+
+        public void Apply(IReceiveEndpointBrokerTopologyBuilder builder)
+        {
+            var consumerName = $"Consumer.{{queue}}.{EntityName}";
```

Comment:
If the only difference is the consumerName, It seems more sensible to have a consumer name formatter, similar to the entity name formatter, which is configurable instead of an entirely different specification.

Reply:
new commits have different implementation